### DISCUSSION TITLE
can `IHttpContext` in asp.netcore be parameter to c++ method?

### DISCUSSION THREAD

**Discussion by ljzj2** [dotnet/aspnetcore#56886] [Q&A]
If I write `.dll` in c++ and there is method:

`__stdcall void Get(IHttpContext context);`

how to use`IHttpContext` ?

**Comment by gfoidl** [comment]
`IHttpContext` is a managed type, thus it can't be used like that. Only blitable types can be used for this to work as intended.

So you need to define a `struct` that holds the relevant properties you'd like to have in native (C++) side too.

For more information see [P/Invoke](https://learn.microsoft.com/en-us/dotnet/standard/native-interop/pinvoke).
